### PR TITLE
To fhir subelement

### PR DESCRIPTION
**Pull request by arscan**
Handles the following cases 

```
addMapping('Patient.name.text', {'identity': 'shr', 'map': '<shr.simple.StringValue>'});
addMapping('Practitioner.active', {'identity': 'shr', 'map': '<shr.simple.DoubleNestedBooleanValue>.<shr.simple.NestedBooleanValue>.<shr.simple.BooleanValue>'});
addMapping('Practitioner.name.text', {'identity': 'shr', 'map': '<shr.simple.NestedStringValue>.<shr.simple.StringValue>'});
```

Note mapping 2 levels deep into name on the FHIR side, and from 2 and 3 levels deep on the shr ES6 side.

I also refactored the fixtures.  Everything in the fhir namespace maps to a fhir resource.  It references various 'simple' elements in the simple namespace.  I'll reuse when I can, but will make new simple elements when necessary.

### CODE REVIEW

File: lib/generateClass.js [hunk 1]
```diff
@@ -301,29 +301,42 @@ function writeToFhir(def, fhir, cw) {
...
+    //   addMapping('Practitioner.active', {'identity': 'shr', 'map': '<shr.simple.DoubleNestedBooleanValue>.<shr.simple.NestedBooleanValue>.<shr.simple.BooleanValue>'});
+    //   addMapping('Practitioner.name.text', {'identity': 'shr', 'map': '<shr.simple.NestedStringValue>.<shr.simple.StringValue>'});
     // }
 
     for (element of fhirProfile.snapshot.element) {
       const mapping = element.mapping && element.mapping.find(m => m['identity'] === 'shr');
 
       if(mapping !== undefined){
-        const match = mapping.map.match(/\<([^\>]*)>/);
+        // const match = mapping.map.match(/\<([^\>]*)>/);
```
Comment (okeefm) [inline lib/generateClass.js]: We should remove this commented line, if the only difference is the `g` signifying a global match.

File: lib/generateClass.js [hunk 2]
```diff
@@ -301,29 +301,42 @@ function writeToFhir(def, fhir, cw) {
...
+        const match = mapping.map.match(/\<([^\>]*)>/g)
         if(match){
-          const name = match[1].split('.').pop();
-          const fhirName = element.path.split('.')[1]; // Currently only handling most basic case
-          if(name === 'Value'){
+          const name = match.map(e => toSymbol(e.slice(1).split('.').pop().slice(0,-1))).join('.')
+          // const name = match[1].split('.').pop();
```
Comment (okeefm) [inline lib/generateClass.js]: Same here as above (re: commented code).

File: lib/generateClass.js [hunk 3]
```diff
@@ -301,29 +301,42 @@ function writeToFhir(def, fhir, cw) {
...
+    //   addMapping('Practitioner.name.text', {'identity': 'shr', 'map': '<shr.simple.NestedStringValue>.<shr.simple.StringValue>'});
     // }
 
     for (element of fhirProfile.snapshot.element) {
       const mapping = element.mapping && element.mapping.find(m => m['identity'] === 'shr');
 
       if(mapping !== undefined){
-        const match = mapping.map.match(/\<([^\>]*)>/);
+        // const match = mapping.map.match(/\<([^\>]*)>/);
+        const match = mapping.map.match(/\<([^\>]*)>/g)
```
Comment (okeefm) [inline lib/generateClass.js]: Why are two of the lesser-than/greater-than characters escaped, while the last one isn't? I'm not sure any of them need to be, based on some quick testing.

File: lib/generateClass.js [hunk 4]
```diff
@@ -301,29 +301,42 @@ function writeToFhir(def, fhir, cw) {
...
             if (def.value instanceof IdentifiableValue && def.value.identifier.isPrimitive) {
-              generateAssignmentIfListFHIR(def.value.card, fhirName, toSymbol(name), cw);
+              generateAssignmentIfListFHIR(def.value.card, fhirName, name, cw);
             } else {
               console.error('ERROR: Value referenced in mapping but none exist on this element.');
             }
           } else {
-            const field = def.fields.find(f => f.identifier.name === name);
-            generateAssignmentIfListFHIR(field.card, fhirName, toSymbol(name), cw);
+            const field = def.fields.find(f => f.identifier.name === match[0].split('.').slice(-1)[0].slice(0,-1));
```
Comment (okeefm) [inline lib/generateClass.js]: Might it be worth moving this bit of code out into its own function? It's used in two places in this snippet, and is non-intuitive if you don't know what you're looking at. Moving it to a function that can be appropriately named/commented might make it a little more clear.

**Comment by arscan** [comment]
I agree, that code was gross.  Updated it to be more readable.

**Pull request merged**